source/js/nav.js

```diff
@@ -1,20 +1,32 @@
 class Nav {
     constructor() {
-        this.setListeners();
+        this.hasMainMenu();
+    }
+
+    hasMainMenu() {
+        document.querySelector('#main-menu') && this.setListeners();
```

Comment:
hasMainMenu does two things, check if menu exists and set listeners. hasMenuMenu should return bool (true/false) and constructor can set the listeners         this.hasMainMenu() && this.setListeners() in order to achieve single responsibillity